gmscompat: Fix permission denial for GCM broadcasts

PeterCxy opened the pull request:
When sending GCM broadcasts, GMS will try to grant temporary power
management exemption to the recipient app. This triggers a
SecurityException and crashes the calling process.

> java.lang.SecurityException: Permission Denial:
> com.google.android.c2dm.intent.RECEIVE broadcast from
> com.google.android.gms (pid=21608, uid=1010132) requires
> android.permission.CHANGE_DEVICE_IDLE_TEMP_WHITELIST or
> android.permission.START_ACTIVITIES_FROM_BACKGROUND or
> android.permission.START_FOREGROUND_SERVICES_FROM_BACKGROUND

Fix this by clearing the exemptions from the broadcast options.

Change-Id: Ie6c60c56350261409233411f2f85d0c9a6d08546

kdrag0n commented:
Thanks! I can merge this now if you rebase — just pushed many gmscompat updates for February yesterday.

PeterCxy committed:
gmscompat: Fix permission denial for GCM broadcasts

When sending GCM broadcasts, GMS will try to grant temporary power
management exemption to the recipient app. This triggers a
SecurityException and crashes the calling process.

> java.lang.SecurityException: Permission Denial:
> com.google.android.c2dm.intent.RECEIVE broadcast from
> com.google.android.gms (pid=21608, uid=1010132) requires
> android.permission.CHANGE_DEVICE_IDLE_TEMP_WHITELIST or
> android.permission.START_ACTIVITIES_FROM_BACKGROUND or
> android.permission.START_FOREGROUND_SERVICES_FROM_BACKGROUND

Fix this by clearing the exemptions from the broadcast options.

Change-Id: Ie6c60c56350261409233411f2f85d0c9a6d08546

PeterCxy commented:
@kdrag0n Rebased to latest tip of sc branch

muhomorr commented:
This issue was fixed recently by stubbing out `setTemporaryAppAllowlist()`:
https://github.com/ProtonAOSP/android_frameworks_base/blob/bbce77d8185f8708a37e4cd50864c57f65f34655/core/java/android/app/BroadcastOptions.java#L171

PeterCxy commented:
@muhomorr Thanks, didn't notice that change at all. Closing this PR then.
